app/feature/insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/InsuranceViewModel.kt

```diff
@@ -0,0 +1,201 @@
+package com.hedvig.android.feature.insurances
+
+import android.annotation.SuppressLint
+import android.net.Uri
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshots.Snapshot
+import arrow.core.raise.either
+import arrow.fx.coroutines.parZip
+import com.hedvig.android.core.common.RetryChannel
+import com.hedvig.android.core.common.android.i
+import com.hedvig.android.core.ui.MoleculeViewModel
+import com.hedvig.android.feature.insurances.data.GetCrossSellsUseCase
+import com.hedvig.android.feature.insurances.data.GetInsuranceContractsUseCase
+import com.hedvig.android.feature.insurances.data.isTerminated
+import com.hedvig.android.notification.badge.data.crosssell.card.CrossSellCardNotificationBadgeService
+import giraffe.InsuranceContractsQuery
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.first
+import octopus.CrossSalesQuery
+
+internal class InsuranceViewModel(
+  private val getInsuranceContractsUseCase: GetInsuranceContractsUseCase,
+  private val getCrossSellsUseCase: GetCrossSellsUseCase,
+  private val crossSellCardNotificationBadgeService: CrossSellCardNotificationBadgeService,
+) : MoleculeViewModel<InsuranceScreenEvent, InsuranceUiState>() {
```

Comment:
This is this https://github.com/cashapp/molecule with some video about it here https://www.youtube.com/watch?v=q9p4ewk-9E4.
I wanted to bring it up in the code camp we got soon, so I thought what perfect way to give it a shot in the app too (I had tried it in sample projects before) so that we got something to reference in the code camp.